Add a section regarding companion tools to the readme (#236)
samply/beam: README.md

This also allows us to detected invalid connection states such as multiple proxies with the same proxy id connecting simultaneously.
In that case the second proxy trying to connect will receive a 409 status code and shut down.

## Companion Tools

Samply.Beam provides a somewhat low-level and focused feature set by design, following the Unix philosophy "Write programs that do one thing and do it well." (following the phrasing of Peter H. Salus). However, Samply.Beam is designed in a way, so that "companion tools" can build upon those functionalities.

> Here are some of the tools to be used with Samply.Beam by the Samply development group:

Reviewer: ```suggestion
Here are some of the companion tools building on Samply.Beam:
```

> - [Beam.Connect](https://github.com/samply/beam-connect) is an HTTP proxy to tunnel arbitrary HTTP calls and APIs through Samply.Beam. It provides fine-grained routing and access control.

Reviewer: ```suggestion
- [Beam.Connect](https://github.com/samply/beam-connect) emulates an HTTP proxy to tunnel arbitrary HTTP calls and APIs through Samply.Beam. It provides fine-grained routing and access control.
```
